app/experimenter/visualization/api/v1/views.py

```diff
@@ -0,0 +1,40 @@
+import json
+
+from django.core.files.storage import default_storage
+from django.http import HttpResponse
+from django.views import View
+
+from experimenter.experiments.models import Experiment
+
+
+class VisualizationView(View):
+    def load_data_from_gcs(self, filename):
+        return (
+            json.loads(default_storage.open(filename).read().decode("utf8"))
+            if default_storage.exists(filename)
+            else []
+        )
+
+    def get(self, request, *args, **kwargs):
+        experiment_json = {}
+
+        slug = self.kwargs["slug"]
+        current_expermient_arr = Experiment.objects.get_prefetched().filter(slug=slug)
```

Comment:
`get_prefetched` is a convenience method to automatically join on all the related tables, but if you only need `recipe_slug` then I don't think this is necessary?

Reply:
You can also use [`get_object_or_404`](https://docs.djangoproject.com/en/3.1/topics/http/shortcuts/#get-object-or-404) here to properly raise 404 if the slug isn't found